### DISCUSSION TITLE
Python connection to milvus 2.4.6 version error

### DISCUSSION THREAD

**Discussion by cnaonaol** [milvus-io/milvus#35417] [Q&A and General discussion]
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=this version of sdk is incompatible with server, please downgrade your sdk or upgrade your server)>

My version：

pymilvus 2.4.4
milvus 2.4.6

**Comment by yhmo** [comment]
This exception only happens when the server and client is incompatible.
Double-check your pymilvus version, maybe the python environment is wrong.
Double-check your milvus version, maybe there are multiple milvus servers deployed.

**Reply by cnaonaol** [reply]
My milvus is a distributed version built using Helm. Pymilvus checked the development documentation and found that milvus 2.4.6 is compatible with Pymilvus 2.4.4, but it still reports this error.

**Comment by xiaofan-luan** [comment]
if it's milvus 2.4.6,it's probably you gonna to hit this issue, can youd double check on it?

**Reply by cnaonaol** [reply]
Yes, I have carefully checked the milvus and pymilvus versions many times. I thought it was a compatibility issue with the latest version, so I have downgraded the milvus version to 2.4.3. After consulting the official documentation, milvus 2.4.3 is compatible with pymilvus 2.4.3. I have done so, but I still encountered this problem.

**Comment by cnaonaol** [comment]
![企业微信截图_17235131006906](https://github.com/user-attachments/assets/807842d6-5c3d-40db-ab5a-070876378802)
![企业微信截图_17235131773906](https://github.com/user-attachments/assets/c6eb27e6-06f9-4847-b10c-443313529470)
This is the error code and error message

**Reply by yhmo** [reply]
I just tried milvus cluster v2.4.6 with pymilvus 2.4.4 and pymivus 2.4.3, I didn't see any problem.
![Screenshot from 2024-08-13 14-58-01](https://github.com/user-attachments/assets/0f070471-4c73-48cc-a5ee-e2f7bed9cc6d)

**Reply by ToanLyHoa** [reply]
Have you solved this problem?

**Reply by xiaofan-luan** [reply]
> Have you solved this problem?

This usually means you are using a pymilvus SDK with higher version than server.

We would recommend you to upgrade your milvus server is possible
